src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/ClaimantResponseCuiCallbackHandler.java

```diff
@@ -62,11 +75,13 @@ private CallbackResponse aboutToSubmit(CallbackParams callbackParams) {
     }
 
     private void updateClaimEndState(AboutToStartOrSubmitCallbackResponse.AboutToStartOrSubmitCallbackResponseBuilder response, CaseData updatedData) {
-
         if (updatedData.hasClaimantAgreedToFreeMediation()) {
             response.state(CaseState.IN_MEDIATION.name());
+        } else if (updatedData.hasApplicantRejectedRepaymentPlan() && (updatedData.getRespondent1().isCompany() || updatedData.getRespondent1().isOrganisation())) {
```

Comment:
any chance to put the condition to separate function?

Reply:
I have grouped the second part to make it shorter and more understandable.